packages/js/product-editor/src/components/editor/editor.tsx

```diff
@@ -41,30 +45,42 @@ type EditorProps = {
 };
 
 export function Editor( { product, settings }: EditorProps ) {
+	const { updateLayoutPath } = useLayoutContext();
+
 	return (
-		<StrictMode>
-			<EntityProvider kind="postType" type="product" id={ product.id }>
-				<ShortcutProvider>
-					<FullscreenMode isActive={ false } />
-					<SlotFillProvider>
-						<InterfaceSkeleton
-							header={ <Header productName={ product.name } /> }
-							content={
-								<>
-									<BlockEditor
-										settings={ settings }
-										product={ product }
-									/>
-									{ /* @ts-expect-error 'scope' does exist. @types/wordpress__plugins is outdated. */ }
-									<PluginArea scope="woocommerce-product-block-editor" />
-								</>
-							}
-						/>
+		<LayoutContextProvider
+			value={ updateLayoutPath( 'product-block-editor' ) }
```

Comment:
Since this would run on every render, should we call `updateLayoutPath` inside a `useEffect` or `useMemo`?

Reply:
Yeahhh, most likely. Since this function is idempotent and non-complex I liked the elegance of being able to plug the helper function directly into the context, but it would be more efficient this way. I'll make the change.

Reply:
I do wonder if this is necessary given some things I've read recently regarding [whether you really need an effect](https://react.dev/learn/you-might-not-need-an-effect#caching-expensive-calculations), which seems relevant to this kind of thing. Curious as to the thoughts of others on the team, but I'll commit this change for now.

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/37720/commits/6e5e0be29882fb42696e5e71a4fa2429483675fc